lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1443,7 +1443,7 @@ def _find_available_certificates(self):
                             logger.debug(
                                 "Secret %s with correct certificate already exists", secret_label
                             )
-                            return
+                            continue
```

Comment:
I think we should add a test for this use case so that we do not reintroduce it later.